code/modules/mob/living/simple_animal/hostile/mimic.dm

```diff
@@ -165,7 +165,7 @@ var/global/list/protected_objects = list(/obj/structure/table, /obj/structure/ca
 			move_to_delay = 2 * I.w_class + 1
 		maxHealth = health
 		if(creator)
-			creator = creator
+			src.creator = creator

```

Comment:
Change the proc argument's name to something less stupid, so you can do "creator = XYZ"